### PR TITLE
Added `no_std` support state transition related crates

### PR DESCRIPTION
**Pull request by xgreenx**
Added `no_std` support state transition related crates:
- `fuel-core-types`
- `fuel-core-storage`
- `fuel-core-executor`

## Checklist
- [x] Breaking changes are clearly marked as such in the PR description and changelog

### Before requesting review
- [x] I have reviewed the code myself

### CODE REVIEW

File: crates/storage/src/kv_store.rs [hunk 1]
```diff
@@ -4,12 +4,27 @@ use crate::{
...
 
 /// The key of the storage.
 pub type Key = Vec<u8>;
+#[cfg(feature = "std")]
 /// The value of the storage. It is wrapped into the `Arc` to provide less cloning of massive objects.
-pub type Value = Arc<Vec<u8>>;
+pub type Value = std::sync::Arc<Vec<u8>>;
+
+#[cfg(not(feature = "std"))]
+/// The value of the storage. It is wrapped into the `Arc` to provide less cloning of massive objects.
```
Comment (MitchTurner) [inline crates/storage/src/kv_store.rs:25]: ]
```suggestion
/// The value of the storage. It is wrapped into the `Rc` to provide less cloning of massive objects.
```

**Review by MitchTurner — DISMISSED** [review]
Approve with one little comment fix.

File: crates/storage/src/vm_storage.rs [hunk 2]
```diff
@@ -58,8 +58,19 @@ use fuel_vm_private::{
 };
 use itertools::Itertools;
 use primitive_types::U256;
+
+#[cfg(feature = "std")]
 use std::borrow::Cow;
 
+#[cfg(not(feature = "std"))]
+use alloc::borrow::Cow;
```
Comment (Dentosal) [inline crates/storage/src/vm_storage.rs:66]:
```suggestion
use alloc::borrow::Cow;
```

nit: I'm pretty sure you can just do this instead

**Review by Dentosal — APPROVED** [review]

File: crates/storage/src/vm_storage.rs (hunk 2 above)
Comment (xgreenx) [reply crates/storage/src/vm_storage.rs:66]: It will work if `alloc` feature is enabled

**Review by Dentosal — APPROVED** [review]

**Review by MitchTurner — APPROVED** [review]

**Pull request merged**